src/Transport/Receiving/MessageReceiverNotifier.cs

```diff
@@ -152,11 +153,9 @@ public void Start()
                     return;
                 }
 
-                //- It is raised when the underlying connection closes because of our close operation
-                var messagingException = exceptionReceivedEventArgs.Exception as MessagingException;
-                if (messagingException != null && messagingException.IsTransient)
+                if (exceptionReceivedEventArgs.Exception.IsTransientException(exceptionReceivedEventArgs.Action))
```

Comment:
Not hang up on this but I would have gone for an extension method that is applied to `exceptionReceivedEventArgs` instead of the current proposal because it makes the code less verbose